v2/pkg/protocols/common/hosterrorscache/hosterrorscache_test.go

```diff
@@ -10,18 +10,69 @@ import (
 func TestCacheCheckMarkFailed(t *testing.T) {
 	cache := New(3, DefaultMaxHostsCount)
 
-	cache.MarkFailed("http://example.com:80", fmt.Errorf("no address found for host"))
-	if value, err := cache.failedTargets.Get("http://example.com:80"); err == nil && value != nil {
-		require.Equal(t, 1, value, "could not get correct number of marked failed hosts")
+	tests := []struct {
+		host     string
+		expected int
+	}{
+		{"http://example.com:80", 1},
+		{"example.com:80", 2},
+		{"example.com", 1},
 	}
-	cache.MarkFailed("example.com:80", fmt.Errorf("Client.Timeout exceeded while awaiting headers"))
-	if value, err := cache.failedTargets.Get("example.com:80"); err == nil && value != nil {
-		require.Equal(t, 2, value, "could not get correct number of marked failed hosts")
+
+	for _, test := range tests {
+		normalizedCacheValue := cache.normalizeCacheValue(test.host)
+		cache.MarkFailed(test.host, fmt.Errorf("no address found for host"))
+		failedTarget, err := cache.failedTargets.Get(normalizedCacheValue)
+		require.Nil(t, err)
+		require.NotNil(t, failedTarget)
+
+		value, ok := failedTarget.(*cacheItem)
+		require.True(t, ok)
+		require.EqualValues(t, test.expected, value.errors.Load())
 	}
-	cache.MarkFailed("example.com", fmt.Errorf("could not resolve host"))
-	if value, err := cache.failedTargets.Get("example.com"); err == nil && value != nil {
-		require.Equal(t, 1, value, "could not get correct number of marked failed hosts")
+
+	for i := 0; i < 3; i++ {
+		cache.MarkFailed("test", fmt.Errorf("could not resolve host"))
 	}
+
+	value := cache.Check("test")
+	require.Equal(t, true, value, "could not get checked value")
+}
+
+func TestCacheItemCheckMarkFailedMultipleCalls(t *testing.T) {
```

Comment:
The purpose of this test should be to call the `cache.MarkFailed` multiple times in parallell from various go routines on the same item, and like my previous comment, we need to ensure that the counter remains above the max error threshold.